libretroshare/src/retroshare/rspeers.h

```diff
@@ -476,9 +493,20 @@ class RsPeers
 	 * @param[in] flags service permissions flag
 	 * @return false if error occurred, true otherwise
 	 */
-	virtual bool addFriend( const RsPeerId &sslId, const RsPgpId& gpgId,
+	virtual bool addFriend( const RsPeerId& sslId, const RsPgpId& gpgId,
 	                        ServicePermissionFlags flags = RS_NODE_PERM_DEFAULT ) = 0;
 
+	/**
+	 * @brief Add trusted node pending PGP key
```

Comment:
As said earlier, optional params here do not make sense. This method should *only* be used in a specific case (because it is quite dangerous) so it should expect a static list of parameters.

Reply:
I dont understand when you would add a friend without a name to display it nor an address or a DNS to contact it. Addign the SSL id alone should actually never happen.

Reply:
Adding it only with SSL id would give only the possibility to accept incoming connection attempts, but I agree is not the most desirable, way of adding someone...